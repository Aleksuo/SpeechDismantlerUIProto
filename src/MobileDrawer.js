

class MobileDrawer extends React.Component {
    state = {
      open: false,
    };

    render(){
        const sideListSwipeable = (
			<div>
				<List>
					<ListItem button key={'Home'}>
						<ListItemIcon><HomeIcon /></ListItemIcon>
						<ListItemIcon></ListItemIcon>
						<ListItemText primary={'Home'} />
					</ListItem>
					<ListItem button key={'Statistics'}>
						<ListItemIcon><BarChartIcon /></ListItemIcon>
						<ListItemIcon></ListItemIcon>
						<ListItemText primary={'Statistics'} />
					</ListItem>
					<ListItem button key={'Settings'}>
						<ListItemIcon><BuildIcon /></ListItemIcon>
						<ListItemIcon></ListItemIcon>
						<ListItemText primary={'Settings'} />
					</ListItem>
					<ListItem button key={'About'}>
						<ListItemIcon><InfoIcon /></ListItemIcon>
						<ListItemIcon></ListItemIcon>
						<ListItemText primary={'About'} />
					</ListItem>
				</List>
			</div>
		)
        return(

        )
    }
}