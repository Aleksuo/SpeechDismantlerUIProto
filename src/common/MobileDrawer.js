import React from 'react'
import PropTypes from 'prop-types'
import {SwipeableDrawer} from '@material-ui/core'
import NavigationList from './NavigationList'

class MobileDrawer extends React.Component {
    state = {
        left: false,
    };

    toggleDrawer = (side, open) => () => {
        this.setState({
            [side]: open,
        })
    }

    render() {
        const {setView} = this.props
        const sideListSwipeable = (
            <div>
                <NavigationList setView = {setView}/>
            </div>
        )
        return (
            <div>
                {/*
				<div>
					<Grid container
						spacing={24}
						direction="column"
						alignItems="left"
						justify="space-evenly">
						<Hidden ndUp>
							<Grid item xs={12} md={12}>
								<Button onClick={this.toggleDrawer('left', true)}>OPEN SWIPE</Button>
							</Grid>
						</Hidden>
					</Grid>
				</div>
				*/}
                <SwipeableDrawer
                    open={this.state.left}
                    onClose={this.toggleDrawer('left', false)}
                    onOpen={this.toggleDrawer('left', true)}
                >
                    <div
                        tabIndex={0}
                        role="button"
                        onClick={this.toggleDrawer('left', false)}
                        onKeyDown={this.toggleDrawer('left', false)}
                    >
                        {sideListSwipeable}
                    </div>
                </SwipeableDrawer>
            </div>

        )
    }
}

MobileDrawer.propTypes = {
	setView: PropTypes.func
}

export default MobileDrawer