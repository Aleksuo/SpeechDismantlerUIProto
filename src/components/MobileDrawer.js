import React from 'react'
import {SwipeableDrawer, List, ListItem, ListItemText} from '@material-ui/core'
import ListItemIcon from '@material-ui/core/ListItemIcon'
import InfoIcon from '@material-ui/icons/Info'
import HomeIcon from '@material-ui/icons/Home'
import BarChartIcon from '@material-ui/icons/BarChart'
import BuildIcon from '@material-ui/icons/Build'
import NavigationList from './NavigationList'

class MobileDrawer extends React.Component {
    state = {
        left: false,
    };

    toggleDrawer = (side, open) => () => {
        this.setState({
            [side]: open,
        })
    }

    render() {
        const {setView} = this.props
        const sideListSwipeable = (
            <div>
                <NavigationList setView = {setView}/>
            </div>
        )
        return (
            <div>
                {/*
				<div>
					<Grid container
						spacing={24}
						direction="column"
						alignItems="left"
						justify="space-evenly">
						<Hidden ndUp>
							<Grid item xs={12} md={12}>
								<Button onClick={this.toggleDrawer('left', true)}>OPEN SWIPE</Button>
							</Grid>
						</Hidden>
					</Grid>
				</div>
				*/}
                <SwipeableDrawer
                    open={this.state.left}
                    onClose={this.toggleDrawer('left', false)}
                    onOpen={this.toggleDrawer('left', true)}
                >
                    <div
                        tabIndex={0}
                        role="button"
                        onClick={this.toggleDrawer('left', false)}
                        onKeyDown={this.toggleDrawer('left', false)}
                    >
                        {sideListSwipeable}
                    </div>
                </SwipeableDrawer>
            </div>

        )
    }
}

export default MobileDrawer