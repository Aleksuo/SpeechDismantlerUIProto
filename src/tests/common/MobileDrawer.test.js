/* eslint-disable */
import React from "react"
import Enzyme, {shallow, mount,render} from "enzyme"
import MobileDrawer from "../../common/MobileDrawer"
import Adapter from "enzyme-adapter-react-16"

Enzyme.configure({adapter: new Adapter()})


describe("MobileDrawer component", () =>{
    test("renders", () =>{ 
        const wrapper = shallow(<MobileDrawer/>) 
        expect(wrapper).toMatchSnapshot()
    })
})