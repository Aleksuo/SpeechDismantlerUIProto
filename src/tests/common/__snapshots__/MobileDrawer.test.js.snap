// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileDrawer component renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MobileDrawer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithTheme(SwipeableDrawer)
        onClose={[Function]}
        onOpen={[Function]}
        open={false}
      >
        <div
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <div>
            <NavigationList />
          </div>
        </div>
      </WithTheme(SwipeableDrawer)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <div>
            <NavigationList />
          </div>
        </div>,
        "onClose": [Function],
        "onOpen": [Function],
        "open": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            <NavigationList />
          </div>,
          "onClick": [Function],
          "onKeyDown": [Function],
          "role": "button",
          "tabIndex": 0,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavigationList />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "setView": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithTheme(SwipeableDrawer)
          onClose={[Function]}
          onOpen={[Function]}
          open={false}
        >
          <div
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div>
              <NavigationList />
            </div>
          </div>
        </WithTheme(SwipeableDrawer)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div>
              <NavigationList />
            </div>
          </div>,
          "onClose": [Function],
          "onOpen": [Function],
          "open": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              <NavigationList />
            </div>,
            "onClick": [Function],
            "onKeyDown": [Function],
            "role": "button",
            "tabIndex": 0,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavigationList />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "setView": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": "div",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
